parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -175,12 +188,53 @@ pub mod pallet {
 	pub(super) type ExecutionHeaders<T: Config> =
 		StorageMap<_, Identity, H256, ExecutionHeader, OptionQuery>;
 
+	/// Execution headers ring buffer map implementation
+
+	/// Index storage for execution header ring buffer map
+	#[pallet::storage]
+	pub(crate) type ExecutionHeaderBufferIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
```

Comment:
`u32` should more than sufficient, and save storage costs.

Reply:
Fair enough. Though I do not have access to this PR to make this change.

Reply:
@ParthDesai No worry, actually we have a lot of refactoring recently.  I'll rebase the PR and make change required later.

Reply:
@yrong Thank you. If you need my help, let me know. For my project, we really need this capability.